pkg/ibc/ibc.go

```diff
@@ -0,0 +1,237 @@
+	rpcTimeout     = "10s"
+	keyringBackend = "test"
+)
+
+type ChainData struct {
+	ChainID  string
+	RPCAddr  string
+	ClientID string
+}
+
+type ClientsInfo struct {
+	ChainA                 *relayer.Chain
+	ChainAClientInfo       relayer.ClientStateInfo
+	ChainAClientExpiration time.Time
+	ChainB                 *relayer.Chain
+	ChainBClientInfo       relayer.ClientStateInfo
+	ChainBClientExpiration time.Time
+}
+
+func (ci ClientsInfo) PathName() string {
+	chainAID := ""
+	if ci.ChainA != nil {
+		chainAID = ci.ChainA.ChainID()
+	}
+
+	chainBID := ""
+	if ci.ChainB != nil {
+		chainBID = ci.ChainB.ChainID()
+	}
+
+	return fmt.Sprintf("%s<->%s", chainAID, chainBID)
+}
+
+type RPC struct {
+	ChainID string `yaml:"chainId"`
+	URL     string `yaml:"url"`
+}
+
+type Config struct {
+	RPCs   []RPC `yaml:"rpc"`
+	GitHub struct {
+		Org    string `yaml:"org"`
+		Repo   string `yaml:"repo"`
+		IBCDir string `yaml:"dir"`
+	} `yaml:"github"`
+}
+
+func (c *Config) getRPCs() map[string]string {
+	rpcs := map[string]string{}
+
+	for _, rpc := range c.RPCs {
+		rpcs[rpc.ChainID] = rpc.URL
+	}
+
+	return rpcs
+}
+
+func (c *Config) GetClientsInfos(ibcs []*relayer.IBCdata) []ClientsInfo {
+	num := len(ibcs)
+
+	out := make(chan ClientsInfo, num)
+	defer close(out)
+
+	for i := 0; i < num; i++ {
+		go func(i int) {
+			clientsInfo, err := c.GetClientsInfo(ibcs[i])
+			if err != nil {
+				out <- ClientsInfo{}
+
+				log.Error(err.Error())
+
+				return
+			}
+			out <- clientsInfo
+		}(i)
+	}
+
+	clientsInfos := []ClientsInfo{}
+
+	for i := 0; i < num; i++ {
+		ci := <-out
+		if !cmp.Equal(ci, ClientsInfo{}) {
+			clientsInfos = append(clientsInfos, ci)
+		}
+	}
+
+	return clientsInfos
+}
+
+func (c *Config) GetClientsInfo(ibc *relayer.IBCdata) (ClientsInfo, error) {
+	clientsInfo := ClientsInfo{}
+
+	rpcs := c.getRPCs()
+
+	cdA := ChainData{
+		ChainID:  ibc.Chain1.ChainName,
+		RPCAddr:  rpcs[ibc.Chain1.ChainName],
+		ClientID: ibc.Chain1.ClientID,
+	}
+
+	chainA, err := prepChain(cdA)
+	if err != nil {
+		return ClientsInfo{}, fmt.Errorf("Error: %w for %v", err, cdA)
+	}
+
+	clientsInfo.ChainA = chainA
+
+	cdB := ChainData{
+		ChainID:  ibc.Chain2.ChainName,
+		RPCAddr:  rpcs[ibc.Chain2.ChainName],
+		ClientID: ibc.Chain2.ClientID,
+	}
+
+	chainB, err := prepChain(cdB)
+	if err != nil {
+		return ClientsInfo{}, fmt.Errorf("Error: %w for %v", err, cdB)
+	}
+
+	clientsInfo.ChainB = chainB
+
+	ctx := context.Background()
+
+	clientsInfo.ChainAClientExpiration, clientsInfo.ChainAClientInfo, err = relayer.QueryClientExpiration(ctx, chainA, chainB)
+	if err != nil {
+		return ClientsInfo{}, fmt.Errorf("Error: %w path %v <-> %v", err, cdA, cdB)
+	}
+
+	clientsInfo.ChainBClientExpiration, clientsInfo.ChainBClientInfo, err = relayer.QueryClientExpiration(ctx, chainB, chainA)
+	if err != nil {
+		return ClientsInfo{}, fmt.Errorf("Error: %w path %v <-> %v", err, cdB, cdA)
+	}
+
+	return clientsInfo, nil
+}
+
+func (c *Config) IBCPaths() ([]*relayer.IBCdata, error) {
```

Comment:
Lets separate out config management into a separat package to keep ibc package clean